src/hooks/use-file-upload.tsx

```diff
@@ -81,14 +83,27 @@ export function useFileUpload({
   useEffect(() => {
     if (!dropRef.current) return;
 
-    const handleDragOver = (e: DragEvent) => {
-      e.preventDefault();
-      e.stopPropagation();
+    // Global drag events with counter for robust dragOver state
+    const handleWindowDragEnter = (e: DragEvent) => {
+      if (e.dataTransfer?.types?.includes("Files")) {
+        dragCounter.current++;
+        setDragOver(true);
+      }
     };
-
-    const handleDrop = async (e: DragEvent) => {
+    const handleWindowDragLeave = (e: DragEvent) => {
+      if (e.dataTransfer?.types?.includes("Files")) {
+        dragCounter.current--;
```

Comment:
nit, a little clearer
```suggestion
        dragCounter.current -= 1;
```